### PR TITLE
feat(project): intial implementation

### PR DESCRIPTION
**Pull request by doktordirk**
showcase https://github.com/doktordirk/aurelia-todo-client/tree/config

sort of preview of use within a plugin https://github.com/SpoonX/aurelia-authentication/pull/250
though, importing defaults into config does not do much for you

### CODE REVIEW

**Comment by doktordirk** [comment]
@RWOverdijk has a look

File: src/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  /**
+   * Get a new resolver for `namespace`.
+   *
+   * @param {string} namespace The namespace
+   *
+   * @return {Configuration}  Resolves to the config segement of the namespace
+   */
+  static of(namespace: string): Configuration {
```
Comment (RWOverdijk) [inline src/resolvers.js:41]: Rename to "for". Of is a reserved keyword.

File: src/config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * @extends Homefront
+ * Config class
+ */
+export class Config extends Homefront {
+  constructor() {
+    super();
+
+    this.put('aurelia-config.configure', true);
```
Comment (RWOverdijk) [inline src/config.js:11]: Move to super() call.

File: spoonx.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+
+var appRoot = 'src/';
+
+module.exports = {
```
Comment (RWOverdijk) [inline spoonx.js:4]: What is this file for?

File: src/aurelia-config.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+ * configure function for aurelia-config
+ * @param {FrameworkConfiguration} frameworkConfiguration The FrameworkConfiguration instance
+ * @param {[string|{}]} plugins Array with plugins to load or function(configManager)
+ * @param {...{}} appConfigs List of appConfigs to merge
+ * @return {Promise<>}
+ */
+export function configure(frameworkConfiguration: FrameworkConfiguration, plugins: Array<string|{}>, ...appConfigs): Promise<> {
+  let config = frameworkConfiguration.container.get(Config);
+  let configManager = new ConfigManager(config, frameworkConfiguration.aurelia.loader);
+  frameworkConfiguration.container.registerInstance(ConfigManager, configManager);
```
Comment (RWOverdijk) [inline src/aurelia-config.js:15]: Newline before method call

File: src/aurelia-config.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * configure function for aurelia-config
+ * @param {FrameworkConfiguration} frameworkConfiguration The FrameworkConfiguration instance
+ * @param {[string|{}]} plugins Array with plugins to load or function(configManager)
+ * @param {...{}} appConfigs List of appConfigs to merge
+ * @return {Promise<>}
+ */
+export function configure(frameworkConfiguration: FrameworkConfiguration, plugins: Array<string|{}>, ...appConfigs): Promise<> {
+  let config = frameworkConfiguration.container.get(Config);
+  let configManager = new ConfigManager(config, frameworkConfiguration.aurelia.loader);
```
Comment (RWOverdijk) [inline src/aurelia-config.js:14]: Align consecutive assignments

File: src/resolvers.js (hunk 1 above)
Comment (doktordirk) [reply src/resolvers.js:41]: Öhm, remember Endpoint.of('cx')??

File: spoonx.js (hunk 3 above)
Comment (doktordirk) [reply spoonx.js:4]: that's the spoonx-tools configuration file

File: src/config-manager.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+    let plugin    = plugins.shift();
+    let configure = () => {};
+
+    if (typeof plugin === 'string') {
+      configure = () => frameworkConfiguration.plugin(plugin, this.config.data[plugin]);
+    } else if (typeof plugin === 'object' && plugin.plugin && plugin.configure !== false) {
+      configure = () => frameworkConfiguration.plugin(plugin.plugin, plugin.rootConfig ? this.config.data : this.config.data[plugin.alias || plugin.plugin]);
+    }
+
+    return Promise.resolve(configure()).then(() => this.configurePlugins(frameworkConfiguration, plugins));
```
Comment (RWOverdijk) [inline src/config-manager.js:103]: You can just loop, and concat the .plugin() calls. Aurelia already handles these in order.

Comment (doktordirk) [reply src/config-manager.js:103]: ill do it, but i'm not sure what happens if one of the plugins is async. but then, it still probably should/would behave the same as in the standard plugin configurations.

**Pull request merged**